crates/ruma-common/src/events/ignored_user_list.rs

```diff
@@ -66,15 +68,16 @@ mod tests {
             "type": "m.ignored_user_list"
         });
 
-        assert_matches!(
-            from_json_value::<AnyGlobalAccountDataEvent>(json).unwrap(),
-            AnyGlobalAccountDataEvent::IgnoredUserList(
+        let ignored_users = assert_matches!(
+            from_json_value::<AnyGlobalAccountDataEvent>(json),
+            Ok(AnyGlobalAccountDataEvent::IgnoredUserList(
                 IgnoredUserListEvent {
                     content: IgnoredUserListEventContent {
                         ignored_users
                     },
-                })
-         if ignored_users == vec![user_id!("@carl:example.com")]
+                }
+            )) => ignored_users
         );
+        assert_eq!(ignored_users, vec![user_id!("@carl:example.com")]);
```

Comment:
This can be shortened a lot, to

```rust
        let ev = assert_matches!(
            from_json_value::<AnyGlobalAccountDataEvent>(json),
            Ok(AnyGlobalAccountDataEvent::IgnoredUserList(ev)) => ev
        );
        assert_eq!(ev.content.ignored_users, vec![user_id!("@carl:example.com")]);
```